frontend/src/scenes/funnels/funnelLogic.ts

```diff
@@ -128,7 +145,27 @@ export const funnelLogic = kea<funnelLogicType>({
                     }
                     breakpoint()
                     insightLogic.actions.endQuery(queryId, ViewType.FUNNELS, result.last_refresh)
-                    actions.setSteps(result.result)
+                    actions.setSteps(result.result as FunnelStep[])
+                    let binsResult: FunnelsTimeConversionResult
+                    if (params.display === ChartDisplayType.FunnelsHistogram) {
+                        try {
+                            params.funnel_viz_type = 'time_to_convert'
+                            params.funnel_to_step = values.histogramStep
+                            binsResult = await pollFunnel(params)
+                        } catch (e) {
+                            insightLogic.actions.endQuery(queryId, ViewType.FUNNELS, null, e)
+                            eventUsageLogic.actions.reportFunnelCalculated(
+                                eventCount,
+                                actionCount,
+                                interval,
+                                false,
+                                e.message
+                            )
+                            return []
+                        }
+                        insightLogic.actions.endQuery(queryId, ViewType.FUNNELS, binsResult.last_refresh)
+                        actions.setTimeConversionBins(binsResult.result as number[])
+                    }
```

Comment:
I ended up having to make both API calls because for example the steps dropdown selection contains data that isn't found in the time conversion funnel call (average conversion time and conversion rate for the other steps). 

<img width="551" alt="Screen Shot 2021-07-09 at 8 06 24 PM" src="https://user-images.githubusercontent.com/25164963/125301277-d4822d80-e2f8-11eb-96c4-e24352ffe2b5.png">

Reply:
Ok, makes sense :).

Reply:
that method can definitely be cleaned up though 😭 it's ginormous, shall be on the TODO